### ISSUE TITLE
Unused imports warning

### ISSUE THREAD

**Issue by chyvonomys** [Geal/nom#705]
I get warnings when `expr_opt!` is used inside `do_parse!`.
**Version**: latest nom, `rustc 1.24.0 (4d90ac38c 2018-02-12)`
**Test**:
```rust
#[macro_use]
extern crate nom;

use nom::le_u8;

named!(test<u8>, do_parse!(x: le_u8 >> y: expr_opt!(Some(x)) >> (y)));
    
fn main() {
    let _ = test(&[0u8]).unwrap();
}
```
**Warning**:
```
warning: unused imports: `IResult`, `Needed`
 --> src/main.rs:6:1
  |
6 | named!(test<u8>, do_parse!(x: le_u8 >> y: expr_opt!(Some(x)) >> (y)));
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
```
**Macros expansion**:
```
note: trace_macro
 --> src/main.rs:9:1
  |
9 | named!(test<u8>, do_parse!(x: le_u8 >> y: expr_opt!(Some(x)) >> (y)));
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: expanding `named! { test < u8 > , do_parse ! ( x : le_u8 >> y : expr_opt ! ( Some ( x ) ) >> ( y )
          ) }`
  = note: to `fn test ( i : & [ u8 ] ) -> $crate :: IResult < & [ u8 ] , u8 , u32 > {
          do_parse ! ( i , x : le_u8 >> y : expr_opt ! ( Some ( x ) ) >> ( y ) ) }`
  = note: expanding `do_parse! { i , x : le_u8 >> y : expr_opt ! ( Some ( x ) ) >> ( y ) }`
  = note: to `{
          do_parse ! ( __impl i , x : le_u8 >> y : expr_opt ! ( Some ( x ) ) >> ( y ) )
          }`
  = note: expanding `do_parse! { __impl i , x : le_u8 >> y : expr_opt ! ( Some ( x ) ) >> ( y ) }`
  = note: to `do_parse ! (
          __impl i , x : call ! ( le_u8 ) >> y : expr_opt ! ( Some ( x ) ) >> ( y ) ) ;`
  = note: expanding `do_parse! { __impl i , x : call ! ( le_u8 ) >> y : expr_opt ! ( Some ( x ) ) >> ( y ) }`
  = note: to `{
          use :: std :: result :: Result :: * ; let i_ = i . clone (  ) ; match call ! (
          i_ , le_u8 ) {
          Err ( e ) => Err ( e ) , Ok ( ( i , o ) ) => {
          let x = o ; let i_ = i . clone (  ) ; do_parse ! (
          __impl i_ , y : expr_opt ! ( Some ( x ) ) >> ( y ) ) } , } }`

```

**Comment by Geal** [comment]
hello,
this should be fixed in 79e44e2

**Comment by sunjay** [comment]
There seem to be some more unused imports here on line 609:

https://github.com/Geal/nom/blob/727006bf63872fcf9c101af262c776eff785ec35/src/whitespace.rs#L589-L616

I get tons of warnings whenever I use `alt_complete!` in `ws!`.
